src/main/java/org/jabref/gui/JabRefFrame.java

```diff
@@ -279,41 +277,127 @@ private Void showTrackingNotification() {
         return null;
     }
 
-    public void refreshTitleAndTabs() {
+    /**
+     * Determines the titles of all tabs - and the window title.
+     *
+     * We aim for following in a tab:
+     *
+     *   filename (datbase-mode) – path-fragment – JabRef
```

Comment:
The most important part is that the open program is JabRef. So I would put this first. The database mode is not important enough to be displayed in the window title in my opinion.